capa/features/extractors/dnfile/helpers.py

```diff
@@ -299,22 +327,79 @@ def get_dotnet_unmanaged_imports(pe: dnfile.dnPE) -> Iterator[DnUnmanagedMethod]
         # like kernel32.CreateFileA
         yield DnUnmanagedMethod(token, module, method)
 
-
+def get_nested_class_table(pe):
+    nested_class_table = {}
+    
+    # Used to find nested classes in typedef
+    for rid, nestedclass in iter_dotnet_table(pe, dnfile.mdtable.NestedClass.number):
+        assert isinstance(nestedclass, dnfile.mdtable.NestedClassRow)
+        nested_class_table[nestedclass.NestedClass.row_index] = nestedclass.EnclosingClass.row_index
+
+    return nested_class_table
+
+def typedef_helper(index, nested_class_table, typedef_class_table, typedef_name, name):
+    # Append the current typeref name
+    typedef_name.append(name)
+    
+    while nested_class_table[index] in nested_class_table:
+        name = typedef_class_table[nested_class_table[index]-1].TypeName
```

Comment:
Can we move the `nested_class_table` and `typedef_class_table` resolutions to this function instead of passing these as arguments? This will clean up the caller.

Reply:
I moved the `typedef_class_table` resolution to this function on the latest commit. I use the `nested_class_table` in a check in the caller to determine if this helper function should be called. Moving the resolution for `nested_class_table` here may cause the logic to change such that this helper function will have to be called on each iteration from the caller function to obtain the proper output, rather than avoiding that with the if-statement beforehand. An example of this is seen in line `382`. Would there be a better way to implement this, or is it better we leave it as is?

Reply:
A clean solution here would be to move the logic from https://github.com/mandiant/capa/blob/66f01c06d2aeaed201525dde719b92c447446aaa/capa/features/extractors/dnfile/helpers.py#L383-L387 into the `resolve_nested_typedef_name` function. This benefits the caller as it no longer needs to know specifics of how the full name is retrieved from the `TypeDef` table, only that the `resolve_nested_typedef_name` function returns the full name when called. This does require a function call each time a name needs to be resolved but I think the benefit of the simplified logic outweighs this.

Reply:
Moved this logic, including the `typeref` check, to their respective helper functions. Applied to all parts where this is present. I left `enclosing_and_nested_classes_index_table` untouched for now since, at the moment, it appears the table will have to be resolved in each call to `resolve_nested_typedef_name` if the logic were to be moved there.